interp/scope.go

```diff
@@ -144,7 +144,7 @@ func (s *scope) rangeChanType(n *node) *itype {
 		return nil
 	}
 	switch {
-	case c.typ.cat == chanT:
+	case c.typ.cat == chanT, c.typ.cat == chanRecvT:
```

Comment:
Range over send channel is possible and should be also addressed here.

Reply:
:thinking:  I'm not sure it is supported by go itself.

```
(41 ms) yaegi ⍉ λ cat case15/main.go 
package main

func main() {
	var c chan<- struct{} = make(chan struct{})

	for _ = range c {
		println("success")
		return
	}
}

// Output:
// success
(4 ms) yaegi λ go run case15/main.go
# command-line-arguments
case15/main.go:6:10: invalid operation: range c (receive from send-only type chan<- struct {})
```

So what you're expecting is to have yaegi fail with a clean error message here, isn't it ?

Thanks !

Reply:
Sorry, I was wrong. Range is not permitted by go over send channels. It's for receiving only. So yes, catching use of range over a send channel should be nice.

Thanks